[CMS-519] Make action blocks work with videos

levadadenys opened the pull request:
<!--
  A summary of the change, including any relevant background, motivation, and context.
  If relevant, include a description, screenshots, and/or video of the existing and new behavior.
-->
## [CMS-519] Make action blocks work with videos

* [feat(actionBlock): add video support](https://github.com/code-dot-org/code-dot-org/commit/3a2f89a6da54ebe58455bda2bb94496e45cc75e8)
* [feat(actionBlock/fullWidthActionBlock): add video support](https://github.com/code-dot-org/code-dot-org/commit/bdeddf8e04a349107b4a79699477d3b092a18ac2)
* [chore(actionBlock): add ActionBlockWrapperProps](https://github.com/code-dot-org/code-dot-org/commit/adff45f75ccaea43902a70e2ebf4299810f31686)
* [feat(actionBlock): make Video in ActionBlock work on SSR](https://github.com/code-dot-org/code-dot-org/commit/f2a83be4bafcf37b2acb72c6c51d3fdfbbc6ae63)
* [feat(actionBlock/fullWidthActionBlock): make Video in fullWidthAction…](https://github.com/code-dot-org/code-dot-org/commit/751f82b767ff2750ed3b465833fe3fee90c9b84e)
* [feat(actionBlock): update tests](https://github.com/code-dot-org/code-dot-org/commit/dfa7795825e11c64f109aaabbb967d46d469fd1e)

https://github.com/user-attachments/assets/8b81dad9-9a1b-467e-93c0-913eef392ba5


## Links

<!--
  Links to relevant external resources; ie, specification documents, Jira tickets, related PRs, Honeybadger errors, etc.
-->

- Jira: [CMS-519](https://codedotorg.atlassian.net/browse/CMS-519)

## Testing story

<!--
  Does your change include appropriate tests?
  If so, please describe how the tests included in this PR are sufficient.
  If not, please explain why this change does not need to be tested.
-->

<!-- Other aspects to consider. Delete any sections that are not relevant to your change. -->

## Deployment strategy

## Follow-up work

<!--
  List (ideally with Jira links) any clean-up or technical debt that will be addressed in future work.
-->

## Privacy

<!--
  1.	Does this change involve the collection, use, or sharing of new Personal Data?
  2.	Does this change involve a new or changed use or sharing of existing Personal Data?
-->

## Security

<!-- Link to Jira task(s) where sensitive security issues are discussed privately. -->

## Caching

## PR Checklist:

<!--
  The final step! Before you create your PR, double-check that everything is in order.
  Change [ ] to [X] during creation to check boxes.
-->

- [ ] Tests provide adequate coverage
- [ ] Privacy and Security impacts have been assessed
- [ ] Code is well-commented
- [ ] New features are translatable or updates will not break translations
- [ ] Relevant documentation has been added or updated
- [ ] User impact is well-understood and desirable
- [ ] Pull Request is labeled appropriately
- [ ] Follow-up work items (including potential tech debt) are tracked and linked

moshebaricdo reviewed (approved):
Looks good to me! One wonder I have is if we want to show caption and download button for fallback when the video is in an action block, historically we have left these out but for consistency may be worth doing (thought it'll look rough, especially long captions in 3-column action block rows). @kelbyhawn what do you think?

kelbyhawn commented:
> Looks good to me! One wonder I have is if we want to show caption and download button for fallback when the video is in an action block, historically we have left these out but for consistency may be worth doing (thought it'll look rough, especially long captions in 3-column action block rows). @kelbyhawn what do you think?

Just watched the video and had the exact same thought re: captions! I think we should remove them.

kelbyhawn commented on frontend/packages/component-library/src/actionBlock/types.ts:
```diff
@@ -10,6 +11,12 @@ export interface ActionBlockProps extends HTMLAttributes<HTMLDivElement> {
+  /** Action Block video component. We use this composition here to allow using ActionBlock component for ssr pages.
+   * More context can be found in this slack thread: https://codedotorg.slack.com/archives/C07UW4ED66Q/p1744640489709969
+   * */
+  VideoComponent?: typeof Video;
+  /** ActionBlock video */
+  video?: VideoProps;
```
Why do we need both of these? Could we just have `video` and it renders the `<Video>` component w/ the props without needing to check that it exists too?

levadadenys replied:
Because otherwise it won't work for SSR, unfortunately :( 
More info on that can be found via link to slack thread in the comment for VideoComponent. (https://codedotorg.slack.com/archives/C07UW4ED66Q/p1744640489709969)
We already use same approach for heroBanner

kelbyhawn replied:
Ah got it, that makes sense! Thanks for sharing the Slack thread.

kelbyhawn reviewed (commented):
Have one question and think we shouldn't show captions.

levadadenys commented:
> Looks good to me! One wonder I have is if we want to show caption and download button for fallback when the video is in an action block, historically we have left these out but for consistency may be worth doing (thought it'll look rough, especially long captions in 3-column action block rows). @kelbyhawn what do you think?

I can remove it from storybook, but keep support of all videoProps under the hood. And for contentful - we can add only one prop for video instead of multiple and that will be videoSrc or smth like that. What do you think?

levadadenys commented:
Updated the PR. Removed captions from stories, but keep support of all video props for consistency under the hood.

https://github.com/user-attachments/assets/886ff23e-9d11-40a9-adcc-6e7ba3d3ecb7

kelbyhawn reviewed (approved):
🎉 🎥
